ksql-rest-app/src/test/java/io/confluent/ksql/rest/server/resources/KsqlResourceTest.java

```diff
@@ -515,7 +515,10 @@ public void shouldDistributePersistentQuery() {
 
     // Then:
     verify(commandStore).enqueueCommand(
-        eq("CREATE STREAM S AS SELECT * FROM test_stream;"), isA(CreateStreamAsSelect.class),
+        eq("CREATE STREAM S WITH (REPLICAS = 1, PARTITIONS = 4, KAFKA_TOPIC = 'S') "
```

Comment:
Ah... now, is this the `SELECT *` bug you were alluding to? I don't think this is a bug.

The KSQL schema of CT/CS statements, where missing, is _inferred_ by the `KsqlResource` before the command is written to the command topic.  

This means is a schema in the Schema Registry evolves for `{F0}` to `{F0, F2}` the schema of the Stream/Table in the metastore does NOT evolve.

CSAS/CTAS statements build their schemas based on the _KSQL_ schema of the source streams and tables, as they are stored in the metastore. Hence they are decoupled from any schema evolution in the Schema Registry.

So.. no bug, but it still leaves the question... is it a good or bad idea to bake in the schema of the project into CSAS / CTAS statements...   TBH, I'm not sure.  @hjafarpour, @rodesai, @apurvam - what do you think?

Probably the best thing would be to pull this change out into its own PR so that it can be discussed independently of this PR.

Reply:
Good to know it's not a bug! Changing this behavior is not trivial - I looked through the code and maintaining knowledge that a resolved statement came from a `SELECT *` requires a lot of plumbing, though it can be done.

I can either leave this as is (fixing the ROWKEY/ROWTIME issues) or I can open another PR to "fix" this behavior and have it resolve back to `SELECT *`. Thoughts?

Reply:
Fixed #2473